src/operators/takeUntil.ts

```diff
@@ -3,41 +3,56 @@ import Observer from '../Observer';
 import Observable from '../Observable';
 import Subscriber from '../Subscriber';
 
-export default function takeUntil<T>(observable: Observable<any>) {
-  return this.lift(new TakeUntilOperator(observable));
+export default function takeUntil<T>(notifier: Observable<any>) {
+  return this.lift(new TakeUntilOperator(notifier));
 }
 
 class TakeUntilOperator<T, R> implements Operator<T, R> {
-
-  observable: Observable<any>;
-
-  constructor(observable: Observable<any>) {
-    this.observable = observable;
+  constructor(private notifier: Observable<any>) {
   }
 
   call(subscriber: Subscriber<T>): Subscriber<T> {
-    return new TakeUntilSubscriber(subscriber, this.observable);
+    return new TakeUntilSubscriber(subscriber, this.notifier);
   }
 }
 
 class TakeUntilSubscriber<T> extends Subscriber<T> {
-  constructor(destination: Subscriber<T>,
-              observable: Observable<any>) {
-    super(destination);
-    this.add(observable._subscribe(new TakeUntilInnerSubscriber(destination)));
+  private notificationSubscriber: TakeUntilInnerSubscriber<any> = null;
+
+  constructor(public destination: Subscriber<T>,
+              private notifier: Observable<any>) {
+    super(null);
```

Comment:
Sorry, seems I wasn't clear enough. 

Let's assume I write new operator `awesomexxx`. When I compose it, is there way I can determine if inherited subscriber need to handle subscription by itself instead calling subscriber's ctor (`super(destination)`) before actually verify it's behavior by writing subscription test cases? 

I was thinking about this can be some kind of 'composing operator guideline / references' in further.